InputHandler.js

```diff
@@ -0,0 +1,83 @@
+class InputHandler {
+    constructor(elem, options){
+        this.elem = elem;
+        this.input = {
+            keys: [],
+            mouse: [],
+            x: 0,
+            y: 0
+        }
+
+        this.keyUp = this.keyUp.bind(this);
+        this.keyDown = this.keyDown.bind(this);
+
+        this.mouseDown = this.mouseDown.bind(this);
+        this.mouseUp = this.mouseUp.bind(this);
+
+        this.mouseMoveX = this.mouseMoveX.bind(this);
+        this.mouseMoveY = this.mouseMoveY.bind(this);
+        
+        this.startHandler = this.startHandler.bind(this);
+        this.stopHandler = this.stopHandler.bind(this);
+    }
+
+    keyDown(event) {
+        this.input.keys[event.which] = true;
+    }
+
+    keyUp(event) {
+        this.input.keys[event.which] = false;
+    }
+
+    mouseDown(event) {
+        this.input.mouse[event.which] = true;
+    }
+
+    mouseUp(event) {
+        this.input.mouse[event.which] = false;
+    }
```

Comment:
Change to alternative:
event.which is deprecated
see [API](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/which)